framework/auth/core.py

```diff
@@ -574,7 +574,8 @@ def register(self, username, password=None):
         self.date_confirmed = dt.datetime.utcnow()
         self.update_search()
         self.update_search_nodes()
-
+        from website.project import new_public_files_collection
+        new_public_files_collection(self)
         # Emit signal that a user has confirmed
         signals.user_confirmed.send(self)
```

Comment:
It seems like a better approach might be to create the public files collection in a function that listens for this signal. There are multiple ways that user onboarding can occur (currently just this and conferences), so that might be a more future-proof implementation.